tests/phpunit/TestFramework/PhpUnit/Coverage/IndexXmlCoverageParserTest.php

```diff
@@ -273,31 +220,60 @@ public function noCoveredLineReportProviders(): iterable
         ];
     }
 
+    private static function getXml(): string
+    {
+        if (self::$xml !== null) {
+            return self::$xml;
+        }
+
+        $xml = file_get_contents(XmlCoverageFixtures::FIXTURES_COVERAGE_DIR . '/index.xml');
+
+        // Replaces dummy source path with the real path
+        self::$xml = preg_replace(
+            '/(source=\").*?(\")/',
+            sprintf('$1%s$2', realpath(XmlCoverageFixtures::FIXTURES_SRC_DIR)),
+            $xml
+        );
+
+        return self::$xml;
+    }
+
     /**
-     * @param iterable<string, XmlCoverageFixture> $coverageFixtures
-     * @param iterable<string, SourceFileData> $sourceFilesData
+     * @param iterable<XmlCoverageFixture> $coverageFixtures
+     * @param iterable<SourceFileInfoProvider> $sourceFilesData
      */
     private function assertCoverageFixtureSame(
         iterable $coverageFixtures,
         iterable $sourceFilesData
     ): void {
-        $normalizedFixtures = [];
+        $this->assertSame([], array_diff(
+            // Fixtures are not expected to be in any particular order
+            iterator_to_array(self::xmlCoverageFixturesToList($coverageFixtures), false),
+            iterator_to_array(self::sourceFileInfoProvidersToList($sourceFilesData), false),
+        ));
+    }
 
-        foreach ($coverageFixtures as $fixture) {
-            /* @var XmlCoverageFixture $fixture */
-            $normalizedFixtures[$fixture->sourceFilePath] = $fixture->serializedCoverage;
+    /**
+     * @param iterable<SourceFileInfoProvider> $sourceFilesData
+     *
+     * @return Traversable<string>
+     */
+    private static function sourceFileInfoProvidersToList(iterable $sourceFilesData): Traversable
+    {
+        foreach ($sourceFilesData as $provider) {
+            yield $provider->provideFileInfo()->getPathname();
         }
+    }
 
-        $normalizedFilesData = [];
-
-        foreach ($sourceFilesData as $data) {
-            /* @var SourceFileData $data */
-            $normalizedFilesData[$data->getSplFileInfo()->getRealPath()] = $data->retrieveCoverageReport();
+    /**
+     * @param iterable<XmlCoverageFixture> $fixtures
+     *
+     * @return Traversable<string>
+     */
+    private static function xmlCoverageFixturesToList(iterable $fixtures): Traversable
```

Comment:
if you pass an iterable is there a point to return a `Traversable` instead of `iterable`? Same for above

Reply:
Yes, because we use `iterator_to_array`, and it wants `Traversable`. It'll still work with `iterable`, but IIRC Psalm wasn't thrilled about it.